pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotSegmentUploadRestletResource.java

```diff
@@ -270,14 +287,14 @@ public void uploadSegmentAsJson(String segmentJsonStr,    // If segment is prese
       @ApiParam(value = "Whether to enable parallel push protection") @DefaultValue("false") @QueryParam(FileUploadDownloadClient.QueryParameters.ENABLE_PARALLEL_PUSH_PROTECTION) boolean enableParallelPushProtection,
       @Context HttpHeaders headers, @Context Request request, @Suspended final AsyncResponse asyncResponse) {
     try {
-      asyncResponse.resume(uploadSegmentInternal(null, segmentJsonStr, enableParallelPushProtection, headers, request));
+      asyncResponse.resume(uploadSegmentInternal(null, segmentJsonStr, enableParallelPushProtection, headers, request, null));
     } catch (Throwable t) {
       asyncResponse.resume(t);
     }
   }
 
   private SuccessResponse uploadSegmentInternal(FormDataMultiPart multiPart, String segmentJsonStr,
-      boolean enableParallelPushProtection, HttpHeaders headers, Request request) {
+      boolean enableParallelPushProtection, HttpHeaders headers, Request request, FormDataMultiPart segmentMetadata) {
```

Comment:
Why do we need two FormDataMultippart arguments?
We should know from 'request' whether this is a metadata update or segment update.  If we don't, then we can add a boolean saying that the multiPart has metadata instgead of segment data.